server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -125,14 +124,7 @@ public void start()
               intervals
           );
 
-          return intervals
-              .stream()
-              .flatMap((Interval interval) -> timeline.lookup(interval).stream())
-              .flatMap(timelineObjectHolder -> {
-                return StreamSupport.stream(timelineObjectHolder.getObject().payloads().spliterator(), false);
-              })
-              .distinct()
-              .collect(Collectors.toList());
+          return new ArrayList<>(timeline.iterateAllObjects());
```

Comment:
This can result in a different set of holders. `lookup()` can adjust the first and last holders so that they are aligned with the given interval.

Reply:
It doesn't matter because the result of this method is a collection of objects, not holders.